### ISSUE TITLE
Build fails with GNU GCC compiler

### ISSUE THREAD

**Issue by rw-spot** [gorakhargosh/watchdog#288]
OSX aliases 'gcc' to 'clang'. This flag in setup.py is valid with clang, but not with gcc-4.9:

```
# required w/Xcode 5.1+ and above because of '-mno-fused-madd'
'-Wno-error=unused-command-line-argument-hard-error-in-future'
```

Please fix setup.py so that a specific C compiler is not assumed.

**Comment by tamland** [comment]
Some of you please provide a patch that works with both gcc and clang.

**Comment by jseabold** [comment]
Shouldn't the onus be on clang users to add an ARCHFLAGS to fix the clang issue?

**Comment by jseabold** [comment]
E.g., clang users (for the particular versions of Xcode affected by this) should do

```bash
ARCHFLAGS=-Wno-error=unused-command-line-argument-hard-error-in-future pip install watchdog
```

Rather than everyone else having to download the source and edit `setup.py` right? Is there another workaround for this?

**Comment by asmeurer** [comment]
I tried removing that flag but got other compiler errors when using gcc. The best workaround here is to set `CC=clang` when installing watchdog.

**Comment by asmeurer** [comment]
These were the errors by the way

```
gcc -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -arch x86_64 -DWATCHDOG_VERSION_STRING="0.8.3" -DWATCHDOG_VERSION_MAJOR=0 -DWATCHDOG_VERSION_MINOR=8 -DWATCHDOG_VERSION_BUILD=3 -I/Users/aaronmeurer/anaconda3/envs/_build/include/python3.5m -c src/watchdog_fsevents.c -o build/temp.macosx-10.5-x86_64-3.5/src/watchdog_fsevents.o -std=c99 -pedantic -Wall -Wextra -fPIC
In file included from /usr/include/Availability.h:168:0,
                 from /usr/include/stdio.h:65,
                 from /Users/aaronmeurer/anaconda3/envs/_build/include/python3.5m/Python.h:25,
                 from src/watchdog_fsevents.c:22:
/System/Library/Frameworks/CoreServices.framework/Frameworks/FSEvents.framework/Headers/FSEvents.h:274:38: error: expected ‘,’ or ‘}’ before ‘__attribute__’
   kFSEventStreamCreateFlagFileEvents __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_6_0) = 0x00000010,
                                      ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/FSEvents.framework/Headers/FSEvents.h:418:38: error: expected ‘,’ or ‘}’ before ‘__attribute__’
   kFSEventStreamEventFlagItemCreated __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_6_0) = 0x00000100,
                                      ^
src/watchdog_fsevents.c: In function ‘watchdog_FSEventStreamCallback’:
src/watchdog_fsevents.c:136:63: warning: unused parameter ‘stream_ref’ [-Wunused-parameter]
 watchdog_FSEventStreamCallback(ConstFSEventStreamRef          stream_ref,
                                                               ^
src/watchdog_fsevents.c:141:63: warning: unused parameter ‘event_ids’ [-Wunused-parameter]
                                const FSEventStreamEventId     event_ids[])
                                                               ^
src/watchdog_fsevents.c: In function ‘watchdog_add_watch’:
src/watchdog_fsevents.c:334:30: warning: unused parameter ‘self’ [-Wunused-parameter]
 watchdog_add_watch(PyObject *self, PyObject *args)
                              ^
src/watchdog_fsevents.c: In function ‘watchdog_read_events’:
src/watchdog_fsevents.c:417:32: warning: unused parameter ‘self’ [-Wunused-parameter]
 watchdog_read_events(PyObject *self, PyObject *args)
                                ^
src/watchdog_fsevents.c: In function ‘watchdog_remove_watch’:
src/watchdog_fsevents.c:466:33: warning: unused parameter ‘self’ [-Wunused-parameter]
 watchdog_remove_watch(PyObject *self, PyObject *watch)
                                 ^
src/watchdog_fsevents.c: In function ‘watchdog_stop’:
src/watchdog_fsevents.c:491:25: warning: unused parameter ‘self’ [-Wunused-parameter]
 watchdog_stop(PyObject *self, PyObject *emitter_thread)
                         ^
src/watchdog_fsevents.c: At top level:
src/watchdog_fsevents.c:602:1: warning: missing initializer for field ‘m_slots’ of ‘struct PyModuleDef’ [-Wmissing-field-initializers]
 };
 ^
In file included from /Users/aaronmeurer/anaconda3/envs/_build/include/python3.5m/Python.h:92:0,
                 from src/watchdog_fsevents.c:22:
/Users/aaronmeurer/anaconda3/envs/_build/include/python3.5m/moduleobject.h:76:28: note: ‘m_slots’ declared here
   struct PyModuleDef_Slot* m_slots;
                            ^
error: command 'gcc' failed with exit status 1
```

**Comment by BoboTiG** [comment]
Should be fixed with #467.
